src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,131 @@
+                .newBuilder()
+                .setPublicTransport(
+                    RequestsCommon.PublicTransportDetails
+                        .newBuilder()
+                        .setWalkingTimeToStation(this.walkingTimeToStation)
+                )
+                .setTypeValue(this.type.getCode())
+                .build();
+        }
     }
 
+    @Builder
     @Getter
-    @AllArgsConstructor
-    class Custom implements Transportation {
-        private final String value;
-        private final Integer code;
+    class DrivingAndPublicTransport implements Transportation {
+        public final TransportationType type = TransportationType.Types.DRIVING_AND_PUBLIC_TRANSPORT;
+
+        /**
+         * limits the possible duration of walking paths must be > 0 and <= 1800
+         * walkingTimeToStation limit is of low precedence and will not override the global travel time limit
+         */
+        @Builder.Default
+        private Integer walkingTimeToStation = 1800;
+
+        /**
+         * limits the possible duration of driving paths must be > 0 and <= 1800
+         * drivingTimeToStation limit is of low precedence and will not override the global travel time limit
+         */
+        @Builder.Default
+        private Integer drivingTimeToStation = 1800;
+
+        /**
+         * constant penalty to apply to simulate the difficulty of finding a parking spot.
+         * If parkingTime >= 0: apply the parking penalty when searching for possible paths.
+         * parkingTime penalty cannot be greater than the global travel time limit
+         */
+        @Builder.Default
+        private Integer parkingTime = 300;
+
+        @Override
+        public RequestsCommon.Transportation getProtoMessage() {
+            return RequestsCommon.Transportation
+                .newBuilder()
+                .setDrivingAndPublicTransport(
+                    RequestsCommon.DrivingAndPublicTransportDetails
+                        .newBuilder()
+                        .setWalkingTimeToStation(this.walkingTimeToStation)
+                        .setDrivingTimeToStation(this.drivingTimeToStation)
+                        .setParkingTime(this.parkingTime)
+                )
+                .setTypeValue(this.type.getCode())
+                .build();
+        }
     }
 
-    static Transportation[] values() {
-        return Modes.values();
+    @Value
+    class TransportationWithoutDetails implements Transportation {
+        TransportationType type;
+
+        @Override
+        public RequestsCommon.Transportation getProtoMessage() {
+            return RequestsCommon.Transportation.newBuilder().setTypeValue(this.type.getCode()).build();
+        }
+    }
+
+    interface TransportationType {
+        String getValue();
+
+        Integer getCode();
+
+        @Getter
+        @AllArgsConstructor
+        enum Types implements TransportationType {
+            PUBLIC_TRANSPORT("pt", 0),
```

Comment:
Since we're here, instead of integer constants you could use `RequestsCommon.TransportationType.PUBLIC_TRANSPORT_VALUE` and others.